src/robot/running/model.py

```diff
@@ -743,7 +743,12 @@ def source(self) -> Path:
     @property
     def directory(self) -> Path:
         source = self.source
-        return source.parent if source and source.is_file() else source
+        if not source:
+            return source
+        
+        if source.is_dir():
+            return source
+        return source.parent
```

Comment:
We don't typically use empty lines between functions and for consistency reasons it would be good to remove the empty line here as well. Separate `if`s are fine for me, but it would also be possible to write this as a ternary `if/else` like the original code:
```python
return source if not source or source.is_dir() else source.parent
```